src/pages/CirclesPage/OrgCircles.tsx

```diff
@@ -27,6 +32,26 @@ export const OrgCircles = ({
   const isAdmin = (org: OrgWithCircles) =>
     org.circles.map(c => c.users[0]).some(u => u && isUserAdmin(u));
 
+  const [visibleInNav, setVisibleInNav] = useState<boolean>(
+    org?.members?.[0]?.visible ?? true
+  );
+  const queryClient = useQueryClient();
+  const setOrgVisibilityInNav = async () => {
+    const { update_org_members_by_pk } = await client.mutate(
+      {
+        update_org_members_by_pk: [
+          {
+            pk_columns: { id: org.members[0].id },
+            _set: { visible: !visibleInNav },
+          },
+          { visible: true },
+        ],
+      },
+      { operationName: 'orgCircles_hideFromSideNav' }
+    );
+    setVisibleInNav(prev => update_org_members_by_pk?.visible ?? prev);
+    queryClient.invalidateQueries(QUERY_KEY_NAV);
```

Comment:
this works, but you could also try to just update the cached value to be the new value if you want to be fancy https://tanstack.com/query/latest/docs/react/reference/QueryClient#queryclientsetquerydata

Reply:
I update it by your suggestion. the old implementation had one flaw, if the hidden org is clicked from overview page it won't appear in side nav because it wasn't fetched before